### PR TITLE
Towards dev/financial/#16 Paypal unreliable getting payment processor type

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Second reviewer's commit of #12007 - code standardisation in paypal class to prevent e-notices & missing details in messages

Before
----------------------------------------
PHP notice on some payment submissions (eg. via webform_civicrm) and use of "non-standard" param.

After
----------------------------------------
No PHP notice and uses standard parameter so it works with all payment submissions that use standard methods to populate the payment processor array ($this->_paymentProcessor).

Related: #12091

Technical Details
----------------------------------------

Comments
----------------------------------------
@mattwire if you can confirm this still seems good to you I will merge it. I basically went through your commit & re-did it - removing the problematic line in isSupported & fixing this incorrect change
https://github.com/civicrm/civicrm-core/pull/12091/files#diff-96a0ea19e8ac102ef100c35b3b3d1c31R506

I feel OK about this now (as a reviewer's commit)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@mattwire this is also waiting for re-review from you

File: CRM/Core/Payment.php
Comment (mattwire) [inline CRM/Core/Payment.php]: `* @deprecated (Deprecated parameter but used in some messages).`

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

File: CRM/Core/Payment/PayPalImpl.php
Comment (mattwire) [inline CRM/Core/Payment/PayPalImpl.php]: Yes

**Review by mattwire — CHANGES_REQUESTED** [review]
@eileenmcnaughton Ok, reviewed and tested this.  It's effectively no functional change as we're just replacing string checks with const checks which should be more reliable.  If you could just add the `@deprecated` tag to _processorName then it's ok to merge.

**Pull request merged**
